packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/web-services/socket-session-endpoint.ts

```diff
@@ -0,0 +1,203 @@
+    let str_response: string;
+
+    while (true) {
+      args = [this.currentBlockHeight];
+
+      requestBody = this.createRequestBody(requestData, methodName, args);
+      this.log.debug(`Iroha requestBody: ${requestBody}`);
+
+      response = await this.transaction.transact(requestBody);
+      str_response = String(response.transactionReceipt);
+
+      if (str_response.includes("Query response error")) {
+        if (str_response.includes(`"errorCode":3`)) {
+          this.log.info(
+            `Initial max block height is: ${this.currentBlockHeight}`,
+          );
+          break;
+        } else {
+          this.log.error(`Runtime error caught: ${str_response}`);
+          break;
+        }
+      }
+      this.currentBlockHeight++;
+    }
+  }
+
+  private async monitoringRoutine(baseConfig: any) {
+    let transactionReceipt: any;
+    let next: IrohaBlockProgress;
+
+    const args = [this.currentBlockHeight];
+    const methodName: string = IrohaQuery.GetBlock;
+    this.log.debug(`Current block: ${this.currentBlockHeight}`);
+
+    const requestBody = this.createRequestBody(baseConfig, methodName, args);
+    const response = await this.transaction.transact(requestBody);
+    const str_response = String(response.transactionReceipt);
+
+    if (str_response.includes("Query response error")) {
+      if (str_response.includes(`"errorCode":3`)) {
+        this.log.debug("Waiting for new blocks...");
+      } else {
+        this.log.error(`Runtime error caught: ${str_response}`);
+      }
+    } else {
+      this.log.debug(`New block found`);
+      transactionReceipt = response.transactionReceipt;
+      next = { transactionReceipt };
+      this.socket.emit(IrohaSocketSession.Next, next);
+      this.currentBlockHeight++;
+    }
+  }
+
+  public async startMonitor(monitorOptions: any): Promise<void> {
+    this.log.debug(`${IrohaSocketSession.Subscribe} => ${this.socket.id}`);
+    this.log.info(`Starting monitoring blocks...`);
+
+    this.monitorMode = true;
+    await this.getInitialMaxBlockHeight(monitorOptions.baseConfig);
+
+    this.monitoringInterval = setInterval(() => {
+      this.monitoringRoutine(monitorOptions.baseConfig);
+    }, monitorOptions.pollTime);
+  }
+
+  private async validateMethodName(methodName: string): Promise<boolean> {
+    let isValidMethod = false;
+
+    if (Object.values(IrohaQuery as any).includes(methodName)) {
+      this.log.debug(`Method name: ${methodName} (IrohaQuery) is valid`);
+      isValidMethod = true;
+    }
+    if (Object.values(IrohaCommand as any).includes(methodName)) {
+      this.log.debug(`Method name: ${methodName} (IrohaCommand) is valid`);
+      isValidMethod = true;
+    }
+    return isValidMethod;
+  }
+
+  public async stopMonitor(): Promise<void> {
+    this.log.info(`Stopping monitor...`);
+    this.monitorMode = false;
+    clearInterval(this.monitoringInterval);
+  }
+
+  public async sendAsyncRequest(asyncRequestData: any): Promise<any> {
```

Comment:
@StepniowskiP If it supports both sync and async would it make sense to not call it `sendAsyncRequest` but maybe `sendRequest` ?

Reply:
Sure, renamed and refactored